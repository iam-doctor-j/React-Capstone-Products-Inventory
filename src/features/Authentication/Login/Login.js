import React from 'react';
import { Form, Field, withFormik } from 'formik';
import loading from '../../../assets/loading.svg';
// import { login } from '../authProvider';
import { connect } from 'react-redux';
import { login } from '../AuthenticationSlice';
import Card from 'react-bootstrap/Card';
import * as Yup from 'yup';

const LoginForm = (props) => {
    const {errors, touched, isSubmitting} = props;
    console.log(props);
    return(
        <div className="container">
        <div className="row justify-content-center">
            <Card className="col-12 col-md-8 col-lg-6">
                {/* <h1>Login</h1> */}
                <Card.Body>
                    <Card.Title>
                    <div className="d-flex">
                        <span className="heading-text">Login</span>
                        <a className="ml-auto" onClick={props.history.goBack} style={{cursor: 'pointer', color: 'var(--secondary)'}}><i className="fas fa-arrow-circle-left"></i></a>
                    </div>
                    </Card.Title>
                    <Form>
                        <div className="form-group">
                            <label className="lable-text">Email</label>
                            <Field className="form-control" type="text" name="email" placeholder="Enter Username"/>
                            {touched.email && errors.email && <span style={{color: 'red'}}>{errors.email}</span> }
                        </div>
                        <div className="form-group">
                            <label className="lable-text">Password</label>
                            <Field className="form-control" type="password" name="password" placeholder="Enter password"/>
                            {touched.password && errors.password && <span style={{color: 'red'}}>{errors.password}</span> }
                        </div>
                        <button className="btn btn-secondary" type="submit" disabled={isSubmitting}>{!isSubmitting ? <span>Login</span> : <span>Logging in...</span>}</button>
                    </Form>
                </Card.Body>
            </Card>
        </div>
        </div>
    );
};

const Login = withFormik({
    mapPropsToValues(){
        return {
            email: '',
            password: ''
        }
    },
    validationSchema: Yup.object().shape({
        email: Yup.string().email('Enter a valid email').required('Email is required'),
        password: Yup.string().required('Password is required'),
    }),
    handleSubmit(values, { resetForm, setSubmitting, setErrors, props }) {
        console.log(values);
        props.dispatch(login(values.email, values.password, () => {
            props.history.goBack();
        }));
    }
})(LoginForm);

export default connect(state => {
    return {}
})(Login);