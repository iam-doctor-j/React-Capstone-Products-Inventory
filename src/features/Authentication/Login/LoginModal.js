import React from 'react';
import { Form, Field, withFormik } from 'formik';
import loading from '../../../assets/loading.svg';
// import { login } from '../authProvider';
import { connect } from 'react-redux';
import { login } from '../AuthenticationSlice';
import Modal from 'react-bootstrap/Modal';

const LoginForm = (props) => {
    const {errors, touched, isSubmitting} = props;
    console.log(props);
    return(
        <Modal show={true}>
            <Modal.Header closeButton>
                <Modal.Title>Login</Modal.Title>
            </Modal.Header>
            <Modal.Body>
            <Form>
                <div>
                    <Field type="text" name="email" placeholder="Enter Username"/>
                    {touched.usrname && errors.usrname && <span style={{color: 'red'}}>{errors.usrname}</span> }
                </div>
                <div>
                    <Field type="password" name="password" placeholder="Enter password"/>
                </div>
                <button style={{width: 100, padding: 10}} type="submit" disabled={isSubmitting}>{!isSubmitting ? <span>Login</span> : <img height="10" width="10" src={loading}/>}</button>
            </Form>
            </Modal.Body>
        </Modal>
        // <div>
            
        // </div>
    );
};

const Login = withFormik({
    mapPropsToValues(){
        return {
            email: '',
            password: ''
        }
    },
    handleSubmit(values, { resetForm, setSubmitting, setErrors, props }) {
        console.log(values);
        props.dispatch(login(values.email, values.password, () => {
            props.history.goBack();
        }));
    }
})(LoginForm);

export default connect(state => {
    return {}
})(Login);