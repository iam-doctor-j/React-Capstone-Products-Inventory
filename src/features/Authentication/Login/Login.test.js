import React from 'react';
import { Shallow, shallow, mount } from 'enzyme';
import {Login, LoginForm} from './Login';
import Card from 'react-bootstrap/Card';
import toJson from 'enzyme-to-json';

describe('Test Login using Full DOM rendering', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(<Login/>);
    });

    it('should check snapshot renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('Should render one Card element', () => {
        expect(wrapper.find(Card).length).toEqual(1);
    });

    it('should render inputs correctly', () => {
        // console.log(toJson(wrapper.find(LoginForm).find('input[name="email"]')))
        wrapper.find(LoginForm).find('input[name="email"]').simulate('change', {
            persist: () => {},
            target: {
                name: 'email',
                value: 'jyotirmaya@sahu.com'
            }
        });
        // console.log(toJson(wrapper.find(LoginForm).find('input[name="email"]')))
        const emailValue = wrapper.find(LoginForm).find('input[name="email"]').props().value;
        expect(emailValue).toEqual('jyotirmaya@sahu.com');
    });

    afterEach(() => {
        wrapper.unmount();
    })
});

describe('Test Login using shallow rendering', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Login/>);
    });

    it('should render snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should find the LoginForm', () => {
        expect(wrapper.dive().find(LoginForm).length).toEqual(1);
    });
});