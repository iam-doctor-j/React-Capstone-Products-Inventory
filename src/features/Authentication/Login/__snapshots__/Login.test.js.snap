// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login using Full DOM rendering should check snapshot renders correctly 1`] = `
<WithFormik(LoginForm)>
  <Formik
    handleSubmit={[Function]}
    initialValues={
      Object {
        "email": "",
        "password": "",
      }
    }
    onSubmit={[Function]}
    validationSchema={[Function]}
  >
    <LoginForm
      dirty={false}
      errors={Object {}}
      getFieldHelpers={[Function]}
      getFieldMeta={[Function]}
      getFieldProps={[Function]}
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      handleSubmit={[Function]}
      initialErrors={Object {}}
      initialTouched={Object {}}
      initialValues={
        Object {
          "email": "",
          "password": "",
        }
      }
      isSubmitting={false}
      isValid={true}
      isValidating={false}
      registerField={[Function]}
      resetForm={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      submitCount={0}
      submitForm={[Function]}
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      validateOnMount={false}
      values={
        Object {
          "email": "",
          "password": "",
        }
      }
    >
      <div
        className="container"
      >
        <div
          className="row justify-content-center"
        >
          <Card
            body={false}
            className="col-12 col-md-8 col-lg-6"
          >
            <div
              className="col-12 col-md-8 col-lg-6 card"
            >
              <CardBody>
                <div
                  className="card-body"
                >
                  <CardTitle>
                    <ForwardRef
                      className="card-title"
                    >
                      <div
                        className="card-title h5"
                      >
                        <div
                          className="d-flex"
                        >
                          <span
                            className="heading-text"
                          >
                            Login
                          </span>
                          <a
                            className="ml-auto"
                            onClick={[Function]}
                            style={
                              Object {
                                "color": "var(--secondary)",
                                "cursor": "pointer",
                              }
                            }
                          >
                            <i
                              className="fas fa-arrow-circle-left"
                            />
                          </a>
                        </div>
                      </div>
                    </ForwardRef>
                  </CardTitle>
                  <Form>
                    <form
                      action="#"
                      onReset={[Function]}
                      onSubmit={[Function]}
                    >
                      <div
                        className="form-group"
                      >
                        <label
                          className="label-text"
                        >
                          Email
                        </label>
                        <Field
                          className="form-control"
                          name="email"
                          placeholder="Enter Username"
                          type="text"
                        >
                          <input
                            className="form-control"
                            name="email"
                            onBlur={[Function]}
                            onChange={[Function]}
                            placeholder="Enter Username"
                            type="text"
                            value=""
                          />
                        </Field>
                      </div>
                      <div
                        className="form-group"
                      >
                        <label
                          className="label-text"
                        >
                          Password
                        </label>
                        <Field
                          className="form-control"
                          name="password"
                          placeholder="Enter password"
                          type="password"
                        >
                          <input
                            className="form-control"
                            name="password"
                            onBlur={[Function]}
                            onChange={[Function]}
                            placeholder="Enter password"
                            type="password"
                            value=""
                          />
                        </Field>
                      </div>
                      <button
                        className="btn btn-secondary label-text"
                        disabled={false}
                        type="submit"
                      >
                        <span>
                          Login
                        </span>
                      </button>
                      <br />
                    </form>
                  </Form>
                </div>
              </CardBody>
            </div>
          </Card>
        </div>
      </div>
    </LoginForm>
  </Formik>
</WithFormik(LoginForm)>
`;

exports[`Test Login using shallow rendering should render snapshot 1`] = `
<Formik
  handleSubmit={[Function]}
  initialValues={
    Object {
      "email": "",
      "password": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={[Function]}
>
  <Component />
</Formik>
`;
